app/src/org/commcare/android/framework/SessionActivityRegistration.java

```diff
@@ -65,10 +66,15 @@ public static void registerSessionExpiration() {
         }
     }
 
-    public static void returnToLogin(Context context) {
-        Intent i = new Intent(context.getApplicationContext(), LoginActivity.class);
-        i.setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
-        i.putExtra(LoginActivity.REDIRECT_TO_HOMESCREEN, true);
+    private static void letHomeScreenRedirectToLogin(Context context) {
+        Intent i = new Intent(context.getApplicationContext(), CommCareHomeActivity.class);
+        i.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
```

Comment:
```FLAG_ACTIVITY_CLEAR_TOP``` clears the backstack of activities down to the earliest instance of CommCareHomeActivity, which shouldn't have anything else below it on the backstack because it is the first activity launched.